modules/transport-netty/src/main/java/org/elasticsearch/transport/netty/NettyTransport.java

```diff
@@ -182,19 +184,25 @@ protected void doStart() {
     }
 
     private ClientBootstrap createClientBootstrap() {
-        if (blockingClient) {
-            clientBootstrap = new ClientBootstrap(new OioClientSocketChannelFactory(
-                Executors.newCachedThreadPool(daemonThreadFactory(settings, TRANSPORT_CLIENT_WORKER_THREAD_NAME_PREFIX))));
-        } else {
-            int bossCount = NETTY_BOSS_COUNT.get(settings);
-            clientBootstrap = new ClientBootstrap(
-                new NioClientSocketChannelFactory(
-                    Executors.newCachedThreadPool(daemonThreadFactory(settings, TRANSPORT_CLIENT_BOSS_THREAD_NAME_PREFIX)),
-                    bossCount,
-                    new NioWorkerPool(Executors.newCachedThreadPool(
-                        daemonThreadFactory(settings, TRANSPORT_CLIENT_WORKER_THREAD_NAME_PREFIX)), workerCount),
-                    new HashedWheelTimer(daemonThreadFactory(settings, "transport_client_timer"))));
-        }
+        // this doPrivileged is for SelectorUtil.java that tries to set "sun.nio.ch.bugLevel"
+        AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
```

Comment:
I noticed this one in a few places. SelectorUtil does this in a clinit block, so this means as soon as the class gets loaded, its gonna try to do its evil thing.

So an alternative to having this in several places, to consolidate it and reduce the amount of security code, would be to force it to load via `Class.forName` in the `*Plugin` class itself (this is always explicitly initialized by PluginService/unit tests first). Here is an example that does that: https://github.com/elastic/elasticsearch/blob/master/plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsPlugin.java#L39